design_docs/protostar-architecture.md

```diff
@@ -0,0 +1,63 @@
+# Protostar modular architecture
+
+This document proposes the idea how to modularize Protostar. Goals of the modularization are
+- Allow to introduce better ownerships of the codebase parts
+- Make workflow asynchronous between smaller sub-teams operating on contracts between modules
+- Allow incremental rewrite to Rust
+- Remove compilation responsibility from Protostar codebase
+
+
+
+## General overview
+First decomposition of Protostar will introduce 3 modules
+
+- Protostar Gateway
```

Comment:
This name is a bit unfortunate since it coincides with SW's `Gateway`

Reply:
Why not `Protostar CLI` ?